src/server.py

```diff
@@ -0,0 +1,183 @@
+    response_dict = {
+        "method": ("HTTP/1.1 405 Method Not Allowed\r\n"
+                    "Date: Mon, 23 May 2005 22:38:34 GMT\r\n"
+                    "Server: Apache/1.3.3.7 (Unix) (Red-Hat/Linux)\r\n"
+                    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\r\n"
+                    "Etag: '3f80f-1b6-3e1cb03b'\r\n"
+                    "Accept-Ranges:  none\r\n"
+                    "Content-Length: 438\r\n"
+                    "Connection: close\r\n"
+                    "Content-Type: text/html; charset=UTF-8\r\n"
+                    "\r\n"
+                    "<num bytes of content>"),
+        "version": ("HTTP/1.1 403 Forbidden\r\n"
+                    "Date: Mon, 23 May 2005 22:38:34 GMT\r\n"
+                    "Server: Apache/1.3.3.7 (Unix) (Red-Hat/Linux)\r\n"
+                    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\r\n"
+                    "Etag: '3f80f-1b6-3e1cb03b'\r\n"
+                    "Accept-Ranges:  none\r\n"
+                    "Content-Length: 438\r\n"
+                    "Connection: close\r\n"
+                    "Content-Type: text/html; charset=UTF-8\r\n"
+                    "\r\n"
+                    "<num bytes of content>"),
+        "host": ("HTTP/1.1 417 Expectation Failed\r\n"
+                    "Date: Mon, 23 May 2005 22:38:34 GMT\r\n"
+                    "Server: Apache/1.3.3.7 (Unix) (Red-Hat/Linux)\r\n"
+                    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\r\n"
+                    "Etag: '3f80f-1b6-3e1cb03b'\r\n"
+                    "Accept-Ranges:  none\r\n"
+                    "Content-Length: 438\r\n"
+                    "Connection: close\r\n"
+                    "Content-Type: text/html; charset=UTF-8\r\n"
+                    "\r\n"
+                    "<num bytes of content>"),
+        "format": ("HTTP/1.1 400 Bad Request\r\n"
+                    "Date: Mon, 23 May 2005 22:38:34 GMT\r\n"
+                    "Server: Apache/1.3.3.7 (Unix) (Red-Hat/Linux)\r\n"
+                    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\r\n"
+                    "Etag: '3f80f-1b6-3e1cb03b'\r\n"
+                    "Accept-Ranges:  none\r\n"
+                    "Content-Length: 438\r\n"
+                    "Connection: close\r\n"
+                    "Content-Type: text/html; charset=UTF-8\r\n"
+                    "\r\n"
+                    "<num bytes of content>"),
+    }
+    if key == "OK":
+        response_dict[key] = ("HTTP/1.1 200 OK\r\n"
+            "Date: Mon, 23 May 2005 22:38:34 GMT\r\n"
+            "Server: Apache/1.3.3.7 (Unix) (Red-Hat/Linux)\r\n"
+            "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\r\n"
+            "Etag: '3f80f-1b6-3e1cb03b'\r\n"
+            "Accept-Ranges:  none\r\n"
+            "Content-Length: " + str(len(body)) + "\r\n"
+            "Connection: close\r\n"
+            "Content-Type: text/html; charset=UTF-8\r\n"
+            "\r\n" + body)
```

Comment:
Because if there is no body or content_type being sent to the function ie body=None, content_type=None, this function will through an error when it tries to construct a dict that holds a string value that concatenates by referencing these arguments.